pressbooks/pressbooks-book-directory-fe
# Downgrade to LTS node version

Issue #350 

This PR changes the engine requirement to be a LTS version of node (Node 14). It changes the `package-lock.json` file to be compatible with the downgrade. NPM 7 uses lock file version 2 while 6 uses lock file version 1.
This also addresses changes into the `buildspec_dev` file to install the proper node version while building.

### How to test

To test locally it's required to install Node version 14 (LTS), latest version would be ok.
```
❯ npm -v && node -v
6.14.15
v14.18.0
```

After installing node, it's required to delete `node_modules` directory, to clear all installed dependencies.
`rm -rf node_modules`

Then, install all the dependencies again with the correct node version and run the project to check everything is place.
`npm install & npm run dev`

After that we need to test code build process to make sure it runs properly with the `buildspec_dev` file.
Note: we can't test it before because we have node version locked at 15.

If merged and works, we need to update `buildspec` for production.
